src/Microsoft.OpenApi/Validations/Rules/OpenApiPathsRules.cs

```diff
@@ -60,6 +58,38 @@ public static class OpenApiPathsRules
                     }
                 });
 
+        /// <summary>
+        ///  Replaces placeholders in the path with {}, e.g. /pets/{petId} becomes /pets/{} .
+        /// </summary>
+        /// <param name="path">The input path</param>
+        /// <returns>The path signature</returns>
+        private static string GetPathSignature(string path)
+        {
+            int i = 0;
+
+            while (i < path.Length)
+            {
+                int openBrace = path.IndexOf('{', i);
+
+                if (openBrace < 0)
+                {
+                    return path;
+                }
+
+                int closeBrace = path.IndexOf('}', openBrace);
+
+                if (closeBrace < 0)
+                {
+                    return path;
+                }
+
+                path = path.Substring(0, openBrace + 1) + path.Substring(closeBrace, path.Length - closeBrace);
```

Comment:
```suggestion
                path = path.Substring(0, openBrace + 1) + path.Substring(closeBrace);
```

Wouldn't this be enough?